src/Runner.Worker/Variables.cs

```diff
@@ -165,6 +165,21 @@ public DictionaryContextData ToSecretsContext()
             }
             return result;
         }
+
+        public DictionaryContextData ToVarsContext()
+        {
+            var result = new DictionaryContextData();
+            foreach (var variable in _variables.Values)
+            {
+                // Only add the configuration variables to the vars context.
+                if (!variable.Secret && variable.Name.StartsWith(Constants.Variables.ConfigurationVariablePrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    var name = variable.Name.Replace(Constants.Variables.ConfigurationVariablePrefix, string.Empty);
```

Comment:
```suggestion
                    var name = variable.Name.Substring(Constants.Variables.ConfigurationVariablePrefix.Length);
```
This might be safer since Replace will replace all occurrences, and we really want to trim the prefix, right?

Reply:
Removed the prefix logic. We will be passing the variables as part of the context object.